lib/matplotlib/testing/decorators.py

```diff
@@ -391,6 +392,11 @@ def decorator(func):
         _, result_dir = _image_directories(func)
         old_sig = inspect.signature(func)
 
+        if not all(k in old_sig.parameters for k in {"fig_test", "fig_ref"}):
```

Comment:
I'd tend to go for a simpler `if "fig_test" not in old_sig.parameters or "fig_ref" not in old_sig.parameters:`. Unfortunately, this does fit on a single line.

If you work with collections here, you can at least make a set-based test instead of a list comprehension:

```suggestion
        if not {"fig_test", "fig_ref"}.issubset(old_sig.parameters):
```